lib/ContentTypeParser.js

```diff
@@ -94,7 +95,7 @@ ContentTypeParser.prototype.run = function (contentType, handler, request, reply
 
 function rawBody (request, reply, options, parser, done) {
   var asString = parser.asString
-  var limit = options.limit
+  var limit = parser.bodyLimit
```

Comment:
I'm a bit lost. Should the name of the options be `options.bodyLimit`?

I think the actual implementation here should be something like `options.bodyLimit >= 0 ? options.bodyLimit : parser.bodyLimit`?

Reply:
No. `options.bodyLimit` comes from the root Fastify object:

https://github.com/fastify/fastify/blob/7cb07ce2643f8bbfb66a607e3fac522d2f44acd2/lib/ContentTypeParser.js#L74

Thus, `options.bodyLimit` will _always_ be `>= 0` since the Fastify constructor doesn't allow anything else. So the option needs to be set in the same way as the option to determine if the parser should parse the incoming body as a string or a buffer -- by attaching it to the parser object during its construction.

Reply:
Looks like there's going to be trouble deciding which body limit to use. IMO the per-route limit should have the highest priority (with the per-parser limit having priority over the Fastify constructor's limit) but there's no way to tell the difference between a route limit and the constructor's limit (unless a flag is added to `Context`).
https://github.com/fastify/fastify/blob/9cec944c7162439b04938526107c6331455e342e/fastify.js#L474

Reply:
Feel free to push whatever fixes you like to this branch.

Reply:
I'd really like this to make it into v1.0.0. So if anyone has any ideas, please do submit them; preferably via code.